### PR TITLE
Skip observability controller tests if no SA TLS cert

### PR DESCRIPTION
**Pull request by machadovilaca**
**What this PR does / why we need it**:

Stop tests from failing if setups without SA TLS certificates

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly

**Jira Ticket**:

```jira-ticket
none
```

**Release note**:

```release-note
none
```

### CODE REVIEW

**Review by nunnatsa — COMMENTED** [review]
@machadovilaca we try to avoid uding skip from inside the test. Instead, please add a label. The test should be running with the right labels, according to the specific environment, that is know to th he test user.

Instead of fix, add fail, to let the users know witch labels should be used.

See many examples in the func test code.

**Comment by machadovilaca** [comment]
> @machadovilaca we try to avoid uding skip from inside the test. Instead, please add a label. The test should be running with the right labels, according to the specific environment, that is know to th he test user.
> 
> Instead of fix, add fail, to let the users know witch labels should be used.
> 
> See many examples in the func test code.

@nunnatsa updated
is this what you meant?

File: tests/func-tests/observability_controller_test.go [hunk 1]
```diff
@@ -55,3 +68,13 @@ var _ = Describe("Observability Controller", Label(tests.OpenshiftLabel, "observ
 		})
 	})
 })
+
+func serviceAccountTlsCertPathExists() bool {
+	_, err := os.ReadFile(observability.ServiceAccountTlsCertPath)
```
Comment (nunnatsa) [inline tests/func-tests/observability_controller_test.go:73]: No need to actually read the file. you can use `os.Stat()` for that; e.g.
 ```suggestion
	_, err := os.Stat(observability.ServiceAccountTlsCertPath)
```

File: tests/func-tests/observability_controller_test.go [hunk 2]
```diff
@@ -55,3 +68,13 @@ var _ = Describe("Observability Controller", Label(tests.OpenshiftLabel, "observ
...
+	if errors.Is(err, os.ErrNotExist) {
+		return false
+	}
+	Expect(err).ToNot(HaveOccurred())
```
Comment (nunnatsa) [inline tests/func-tests/observability_controller_test.go:77]: Please avoid using gomega asserion in helper function (in new code...).

Please return the error, and asset it from the caller code; e.g. change the function signature to return bool and error.

**Review by nunnatsa — COMMENTED** [review]
Added a few inline comments

File: tests/func-tests/observability_controller_test.go (hunk 1 above)
Comment (machadovilaca) [reply tests/func-tests/observability_controller_test.go]: updated

File: tests/func-tests/observability_controller_test.go (hunk 2 above)
Comment (machadovilaca) [reply tests/func-tests/observability_controller_test.go]: updated

**Review by nunnatsa — APPROVED** [review]
/approve

**Pull request merged**